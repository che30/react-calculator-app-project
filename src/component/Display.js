import React from 'react';

const Display = (result) => {
  const {
    total, next, operation,
  } = result;
  let currentDisplay;
  if ((total && !operation && !next)) {
    currentDisplay = total;
  } else if (total && operation && !next) {
    currentDisplay = operation;
  } else if (total && operation) {
    currentDisplay = next;
  } else if (!operation && !next && !total) {
    currentDisplay = '0';
  }
  return (
    <div className="Display">
      {currentDisplay}
    </div>
  );
};
export default Display;
