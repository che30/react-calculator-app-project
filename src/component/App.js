import React, { useState } from 'react';
import Buttonpanel from './Buttonpanel';
import Display from './Display';
import Calculate from '../logic/calculate';
import '../index.css';

const App = () => {
  const [calculator, setCalculator] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const handleClick = (buttonName) => {
    const result = Calculate(calculator, buttonName);
    console.log(result);
    setCalculator({ ...result });
  };

  return (
    <div className="main">

      <Display
        total={calculator.total}
        next={calculator.next}
        operation={calculator.operation}
      />

      <Buttonpanel clickHandler={handleClick} />
    </div>
  );
};

export default App;
