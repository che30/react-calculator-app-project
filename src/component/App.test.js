import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
// import regeneratorRuntime from 'regenerator-runtime';
import Display from './Display';

import App from './App';

it('renders correctly', () => {
  const tree = renderer
    .create(<App/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
describe('Calculator operations', () => {
  beforeEach(() => {
    render(<App />);
  });
it('type', () => {
  const buttonEl = screen.getByText(/1/i);
  expect(buttonEl).toBeInTheDocument();
})
it('add two numbers', async ()=>{
  await userEvent.click(screen.getByText('1'));
    await userEvent.click(screen.getByText('2'));
    await userEvent.click(screen.getByText('+'))
    await userEvent.click(screen.getByText('3'));
    await userEvent.click(screen.getByText('4'));
    await userEvent.click(screen.getByText('='));
    const display = await screen.findByText(/46/);
    expect(display).toBeInTheDocument();
})
it('subtract two numbers', async ()=>{
  await userEvent.click(screen.getByText('1'));
    await userEvent.click(screen.getByText('2'));
    await userEvent.click(screen.getByText('-'))
    await userEvent.click(screen.getByText('3'));
    await userEvent.click(screen.getByText('4'));
    await userEvent.click(screen.getByText('='));
    const display = await screen.findByText(/-22/);
    expect(display).toBeInTheDocument();
})
it('Multiply two numbers', async ()=>{
  await userEvent.click(screen.getByText('1'));
    await userEvent.click(screen.getByText('2'));
    await userEvent.click(screen.getByText('*'))
    await userEvent.click(screen.getByText('3'));
    await userEvent.click(screen.getByText('4'));
    await userEvent.click(screen.getByText('='));
    const display = await screen.findByText(/408/);
    expect(display).toBeInTheDocument();
})
it('Divide two numbers', async ()=>{
  await userEvent.click(screen.getByText('2'));
  await userEvent.click(screen.getByText('0'));
  await userEvent.click(screen.getByText('/'))
  await userEvent.click(screen.getByText('2'));
  await userEvent.click(screen.getByText('='));
  const display = await screen.findByText(/10/);
  expect(display).toBeInTheDocument();
})
it('check if +/- is working', async ()=>{
  await userEvent.click(screen.getByText('2'));
  await userEvent.click(screen.getByText('+/-'));
  const display = await screen.findByText(/-2/);
  expect(display).toBeInTheDocument();
})
it('check if AC is working', async ()=>{
  await userEvent.click(screen.getByText('2'));
  await userEvent.click(screen.getByText('3'));
  await userEvent.click(screen.getByText('AC'));
  expect(screen.queryByText('23')).toBeNull();
})


})