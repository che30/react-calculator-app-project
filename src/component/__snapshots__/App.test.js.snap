// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="Display"
  >
    0
  </div>
  <div
    className="buttons-div"
  >
    <div
      className="board-row"
    >
      <button
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        name="/"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
    </div>
    <div
      className="board-row"
    >
      <button
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        name="*"
        onClick={[Function]}
        type="button"
      >
        *
      </button>
    </div>
    <div
      className="board-row"
    >
      <button
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="board-row"
    >
      <button
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="board-row last"
    >
      <button
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
